Edit Release Notes for the Mantid Framework

cailafinn opened the pull request:
### Description of work

#### Summary of work
Editing of framework release notes for v6.13.0.

Refs #39471 

### To test:
1. Build release notes `cmake --build . --target docs-html`. Open in a browser and check for any formatting/spelling issues.


*This does not require release notes* because **it is release notes.**


---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

cailafinn committed:
Edit Algorithms section

RE #39471

cailafinn committed:
Format python release notes

RE #39471

sf1919 commented:
I think this needs to be pointing at `release-next`

sf1919 commented on docs/source/release/v6.13.0/framework.rst:
```diff
@@ -10,61 +10,105 @@ Algorithms
+- New algorithm :ref:`algm-CombineTableWorkspaces` allows combination of a pair of :ref:`Table Workspaces`, provided
+  they have matching column names and data types.
+- Introduced a new python algorithm, :ref:`algm-RefineSingleCrystalGoniometer`, that refines the UB-matrix and
+  goniometer offsets simultaneously. This Improves the indexing of the peaks for those cases when there is sample
```
```suggestion
  goniometer offsets simultaneously. This improves the indexing of the peaks for those cases when there is sample
```

sf1919 commented on docs/source/release/v6.13.0/framework.rst:
```diff
@@ -10,61 +10,105 @@ Algorithms
+  which may have been incorrect under certain conditions. This could lead to incorrect grouping of pixels in the output
+  workspace. The issue was caused by an incorrect handling of the pixel indices when calculating the groups. The fix
+  ensures that pixel indices are correctly handled, resulting in accurate grouping of pixels in the output workspace.
+- In the script generated by :ref:`GeneratePythonFitScript <algm-GeneratePythonFitScript>` titles for the subplots have
```
```suggestion
- In the script generated by :ref:`GeneratePythonFitScript <algm-GeneratePythonFitScript>`, titles for the subplots have
```

sf1919 commented on docs/source/release/v6.13.0/framework.rst:
```diff
@@ -10,61 +10,105 @@ Algorithms
+  ensures that pixel indices are correctly handled, resulting in accurate grouping of pixels in the output workspace.
+- In the script generated by :ref:`GeneratePythonFitScript <algm-GeneratePythonFitScript>` titles for the subplots have
+  been replaced by meaningful legends so that labels of output subplots do not overlap.
+- The 'Unweighted least squares' cost function in :ref:`algm-Fit` now works as expected when compared with the scipy
```
Incorrect formatting on this line

cailafinn replied:
What's incorrect about the formatting here?

sf1919 replied:
The Unweighted least squares

cailafinn replied:
Ah, I didn't realise this was a code quote. I'll change that.

sf1919 commented on docs/source/release/v6.13.0/framework.rst:
```diff
@@ -10,61 +10,105 @@ Algorithms
+
+- ``InstrumentArrayConverter`` and ``PeakData`` utility classes used in peak integration algorithms have been moved into
+  a common module located at ``plugins.algorithms.peakdata_utils.py``. Import paths related to those classes must be
+  updated from ``from plugins.algorithms.IntegratePeaksSkew import InstrumentArrayConverter, PeakData`` into
```
Is this line correct? from seems to be repeated.

To be honest the whole bullet point is difficult to understand. Can @RichardWaiteSTFC help with possibly rewording this?

sf1919 commented on docs/source/release/v6.13.0/framework.rst:
```diff
@@ -95,11 +139,12 @@ New features
 Deprecated
 ############
-- :ref:`LoadPreNexus <algm-LoadPreNexus>` and :ref:`LoadPreNexusMonitors <algm-LoadPreNexusMonitors>` has been deprecated. There is no replacement.
+- :ref:`LoadPreNexus <algm-LoadPreNexus>` and :ref:`LoadPreNexusMonitors <algm-LoadPreNexusMonitors>` has been
```
```suggestion
- :ref:`LoadPreNexus <algm-LoadPreNexus>` and :ref:`LoadPreNexusMonitors <algm-LoadPreNexusMonitors>` have been
```

sf1919 commented on docs/source/release/v6.13.0/framework.rst:
```diff
@@ -119,44 +164,57 @@ Python
 ############
-- Introduced a prototype "side-by-side" help system that includes both the legacy QtHelp-based viewer and a new Python-based Help Window using an embedded web browser (QWebEngine) to display documentation within Mantid Workbench.
-- Improvements:
+- Introduced a prototype "side-by-side" help system that includes both the legacy QtHelp-based viewer and a new
```
I assume this section is a result of the work by @darshdinger. It is not clear from this release note how to access this prototype. Is there relevant documentation that needs to be linked to from this release note? @darshdinger and @peterfpeterson can you advise?

sf1919 replied:
@peterfpeterson queried whether this should be in the release notes for v6.13 at all

cailafinn replied:
Probably not, if we're not intending to enable it this release. I'll move it over to 6.14?

sf1919 reviewed (changes requested):
Thank you for trying to wrangle the Framework release notes. It's a large and complicated section. I've noted a couple of small formatting errors. However there are some release notes that don't make sense and I think we'll need assistance in sorting them out. Hopefully I have linked the right people to those comments.

cailafinn committed:
Fix formatting/spelling issues

RE #39471

cailafinn committed:
Remove new docs system release notes

They're not enabled yet. They'll likely be a improvement visible to
users in v6.14 once the variable is changed in CMake.

RE #39471

sf1919 reviewed (dismissed):
Thank you for making the changes requested and opening the separate PR #39500 . Happy to approve.

cailafinn committed:
Add missing note from diffraction docs

This probably should be in here as well as the diff docs. Committing here so it doesn't get lost. 

RE #39471
